cli/commands/catalog/tui/models/page/model.go

```diff
@@ -178,3 +179,11 @@ func (model Model) footerView() string {
 
 	return lipgloss.JoinVertical(lipgloss.Left, info, model.Buttons.View(), model.keys.View())
 }
+
+// clearScreen - explicit clear screen to avoid terminal hanging
+func clearScreen() tea.Msg {
+	if runtime.GOOS == "darwin" {
+		fmt.Print("\033[H\033[2J\033[3J")
```

Comment:
Where did this come from? Link to docs or some explanation for future maintainers?

Reply:
There are some ANSI commands mentioned on unix.com which I used, added as note in code
https://www.unix.com/os-x-apple-/279401-means-clearing-scroll-buffer-osx-terminal.html